src/main/java/com/group12/LaunchInterceptorConditionCollection.java

```diff
@@ -140,6 +140,83 @@ public boolean LIC3(List<Point> points, double area1) throws IllegalArgumentExce
         return false;
     }
 
+    /**
+     * LIC #6
+     *
+     * @param points points list of radar echos ({@link Point})
+     * @param nPts number of consecutive intervening radar echos
+     * @param dist has to be greater than or equal to zero
+     * @return true if there is a set of <b>nPts</b> consecutive points in <b>points</b> (and points.size() >= 3) s.t.
+     * 1) the distance between the point and the line defined by the first and last points in the set is greater
+     * than <b>dist</b> or 2) the first and last points in the set are considered equal and there is a point
+     * between them at a distance greater than <b>dist</b> from the coinciding point between the first and the last.
+     * false otherwise.
+     * @throws IllegalArgumentException is thrown if <b>points</b> is null, if
+     * <b>nPts</b> is less than 3 or greater than the size of <b>points</b>, or if <b>dist</b> is less
+     * than 0 (zero)
+     */
+
+    public boolean LIC6(List<Point> points, int nPts, double dist) throws IllegalArgumentException {
+        if (points == null) {
+            throw new IllegalArgumentException("Points list cannot be null");
+        }
+
+        // The condition is not met when NUMPOINTS < 3
+        if (points.size() < 3) {
+            return false;
+        }
+
+        // Requirement on nPts is 3 <= N_PTS <= NUMPOINTS
+        if (nPts < 3 || nPts > points.size()) {
+            throw new IllegalArgumentException("Parameter nPts has to be within 3 and points.size() inclusive");
+        }
+
+        // Require DIST to be greater than or equal to zero
+        if (doubleCompare(dist, 0) < 0) {
+            throw new IllegalArgumentException("Dist cannot be less than zero");
+        }
+
+        // For each line defined by the points points[i] and points[i+nPts-1]
+        for (int i = 0; i < points.size() - nPts + 1; i++) {
+
+            // The points {points[i]..points[i+nPts-1]} define a set of nPts points
+            Point linePoint1 = points.get(i);
+            Point linePoint2 = points.get(i + nPts - 1);
+
+            // If linePoint1 and linePoint2 are equal, choose a point for distance calculations which is the mean
+            if (pointsEqual(linePoint1, linePoint2)) {
+                double meanX = linePoint1.getX()/2 + linePoint2.getX()/2;
+                double meanY = linePoint1.getY()/2 + linePoint2.getY()/2;
+
+                Point coincidingPoint = new Point(meanX, meanY);
```

Comment:
I think that if 2 points are completely equal, the middle point of them is still that point?

Like, the middle point of (1,1) and (1,1) is still (1,1). You can just pick `linePoint1`.

Reply:
Because `doubleCompare` has limited precision, I chose to interpret the coinciding point as the point in the middle of point1 and point2. Otherwise, the results would be different depending on whether point1 or point2 was chosen. This can be revised later if required.

Reply:
This makes sense. You can keep it!